Fix #4349: Record leave for refresher exp events

BenHenning opened the pull request:
This is a continuation of #4543. See that and #4349 for context.

NB: I still need to manually test this & consider how to improve automated testing before it's ready for submission.

BenHenning committed:
Clone changes from #4543.

BenHenning committed:
Merge branch 'develop' into event_refresher_exp

BenHenning committed:
Address reviewer comments from #4543 and some additional small changes.

seanlip commented on core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js:
```diff
@@ -656,6 +656,8 @@ oppia.directive('conversationSkin', [
                                   true;
                               }
                             );
+                          ExplorationPlayerService.recordLeaveForRefresherExp(
```
@aks681 found a bug here -- this is in the wrong place. Even if he clicks No in the redirect modal, it's getting recorded.

BenHenning replied:
Thanks; fixed.

BenHenning committed:
Address review comment (moved recordLeaveForRefresherExp to correct
location).

BenHenning committed:
Address final review comments and add fixes to ensure event reporting is
working end-to-end when moving to a refresher exploration.

BenHenning commented:
Thanks; fixed all remaining issues I think. Per Hangouts discussions, updated the modal to ensure:

1. Duplicate clicks does not result in multiple event logs
2. There's enough time before page redirection for the event to be logged

Verified that this works e2e by ensuring there are entities for LeaveForRefresherExplorationEventLogEntryModel after clicking 'Yes' on the refresher redirection modal.

seanlip reviewed (approved):
LGTM

seanlip commented:
Force-merging. Will run e2e tests manually on release branch.
